resources/EPvalidator.rb

```diff
@@ -437,6 +437,11 @@ def self.run_validator(hpxml_doc)
         "TankVolume" => one,
       },
 
+      ## [Have desuperheater]
+      "/HPXML/Building/BuildingDetails/Systems/WaterHeating/WaterHeatingSystem[HasDesuperheater='true']" => {
+        "RelatedHVACSystem" => one,
+      },
```

Comment:
This is good, indeed the "RelatedHVACSystem" requirement should be conditional based on the HasDesuperheater value. But you'll also want to add a requirement to the WaterHeatingSystem like:
`"HasDesuperheater" => zero_or_one, # See [Desuperheater]`
And please rename "[Have desuperheater]" to just "[Desuperheater]".

Reply:
Okay, I added that. Did this requirement really functional, or it's just for pointing to the right location where desuperheater requirements lie.

Reply:
Good question. You're right that the requirement isn't really functional, since `zero_or_one` implies that the element is optional. It's mostly as you said, to point to the right location. When software developers are writing HPXML files that get used by this OS measure, we point them to this EPvalidator.rb file so they can understand what's required. If they are working on a WaterHeatingSystem element, they will now more easily see that they can specify a desuperheater. Previously, it would be easy for someone to not notice the desuperheater section.